db/migrate/20240327000440_replace_post_reply_index.rb

```diff
@@ -0,0 +1,27 @@
+# frozen_string_literal: true
+
+class ReplacePostReplyIndex < ActiveRecord::Migration[7.0]
+  disable_ddl_transaction!
+
+  def up
+    execute <<~SQL
+      CREATE INDEX CONCURRENTLY IF NOT EXISTS "index_posts_on_topic_id_and_reply_to_post_number"
+      ON "posts" ("topic_id", "reply_to_post_number")
+    SQL
```

Comment:
```suggestion
    execute <<~SQL
      DROP INDEX IF EXISTS "index_posts_on_topic_id_and_reply_to_post_number"
    SQL

    execute <<~SQL
      CREATE INDEX CONCURRENTLY IF NOT EXISTS "index_posts_on_topic_id_and_reply_to_post_number"
      ON "posts" ("topic_id", "reply_to_post_number")
    SQL
```

When we add an index concurrently, we need to always drop the index first if it exists because the migration can still fail but leave the index in a corrupted state.